test/test_gen_backend_stubs.py

```diff
@@ -0,0 +1,99 @@
+import os
+import tempfile
+import subprocess
+
+from torch.testing._internal.common_utils import TestCase, run_tests
+
+path = os.path.dirname(os.path.realpath(__file__))
+gen_backend_stubs_path = os.path.join(path, '../tools/codegen/gen_backend_stubs.py')
+
+yaml_with_errors = {
+    # Missing key (backend)
+    '''\
+cpp_namespace: torch_xla
+supported:
+- abs
+    ''': 'AssertionError: You must provide a value for "backend"',
+    # 'backend' key provided but not specified
+    '''\
+backend:
+cpp_namespace: torch_xla
+supported:
+- abs
+    ''': 'AssertionError: You must provide a value for "backend"',
+
+    # Missing key (cpp_namespace)
+    '''\
+backend: XLA
+supported:
+- abs
+    ''': 'AssertionError: You must provide a value for "cpp_namespace"',
+
+
+    # 'cpp_namespace' key provided but not specified (including whitespace)
+    '''\
+backend: XLA
+cpp_namespace:\t
+supported:
+- abs
+    ''': 'AssertionError: You must provide a value for "cpp_namespace"',
+
+    # supported is empty (it should be a list)
+    '''\
+backend: XLA
+cpp_namespace: torch_xla
+supported:
+    ''': 'AssertionError: expected "supported" to be a list, but got: None',
+
+    # supported is a single item (it should be a list)
+    '''\
+backend: XLA
+cpp_namespace: torch_xla
+supported: abs
+    ''': "AssertionError: expected \"supported\" to be a list, but got: abs (of type <class 'str'>)",
+
+    # supported contains an op that isn't in native_functions.yaml
+    '''\
+backend: XLA
+cpp_namespace: torch_xla
+supported:
+- abs_BAD
+    ''': 'AssertionError: Found an invalid operator name: abs_BAD',
+
+    # unrecognized extra yaml key
+    '''\
+backend: XLA
+cpp_namespace: torch_xla
+supported:
+- abs
+invalid_key: invalid_val
+    ''': 'contains unexpected keys: invalid_key. \
+Only the following keys are supported: backend, cpp_namespace, supported, autograd',
+}
+
+# gen_backend_stubs.py is an integration point that is called directly by external backends.
+# The tests here are to confirm that badly formed inputs result in reasonable error messages.
+class TestGenBackendStubs(TestCase):
+
+    def assert_yaml_causes_error(self, yaml_str: str, error_msg: str) -> None:
+        with tempfile.NamedTemporaryFile(mode='w') as fp:
+            fp.write(yaml_str)
+            fp.flush()
+            command = ['python',
+                       gen_backend_stubs_path,
+                       '--output_dir=""',
```

Comment:
So... what does this mean? Dump in CWD?

Reply:
I just got this from the docs- from some examples I assumed that this was the canonical way to pass arguments to a python script. Does that look wrong?

If you agree with the change I suggested in the above comment though, then this doesn't matter

Reply:
What I mean is, you're passing empty string as output dir. So does that mean the script will dump the outputs in the current working directory?

Reply:
yeah, this is a user API thing I should have been more explicit about.

The dry-run flag seems like a reasonable addition to `gen_backend_stubs`, because (a) that allows these tests to run without writing to the file system for no reason, and (b) that makes it more similar to gen.py.

The dry-run flag effectively means that we ignore the output directory though. So as it's set up now, `output_dir=""` doesn't do anything because I've specified the dry_run flag. Another option would be to kill the dry_run flag, and just make the `output_dir` flag optional, where not specifying it does a dry run. I was hesitant about that because I figured that if you're an external backend using this script, you probably don't care about dry-run, and it would be better to have explicit error messages if you pass in wrong flags.